src/api.ts

```diff
@@ -297,16 +296,16 @@ router.post(
       return res.status(400).send(validateBidCancelSchema.errors);
     }
     try {
-      const bidData: Bid | null = await database.getBidBySignedMessage(
+      let bidData = await database.getBidBySignedMessage(
         req.body.bidMessageSignature
-      );
+      ) as CanceledBid;
```

Comment:
This is an unsafe cast and circumvents typescript type safety.

This data type is `Bid | null` not `CanceledBid`, later in the code I see you setting `.cancelDate = timeStamp` which will fill in the `cancelDate` field, but up until that line this is incorrectly typed and therefore dangerous.

Create a new variable later instead of unsafely casting this one, ie:
```
const canceledBid: CanceledBid = {...bidData, cancelDate: timeStamp};
```
This is the proper way of converting.